mass/ports/outbound/dao.py

```diff
@@ -14,11 +14,28 @@
 # limitations under the License.
 #
 
-"""ResourceDao Port"""
+"""ResourceDao Port and DaoCollectionPort"""
+from abc import ABC, abstractmethod
 
 from hexkit.protocols.dao import DaoNaturalId
 from typing_extensions import TypeAlias
 
 from mass.core.models import Resource
 
 ResourceDaoPort: TypeAlias = DaoNaturalId[Resource]
```

Comment:
This is used by the DaoCollectionPort but is not a port itself. Maybe just rename to:
```suggestion
ResourceDao: TypeAlias = DaoNaturalId[Resource]
```